### PR TITLE
OEL-1132: OE Contact Forms integration

### PR DESCRIPTION
**Pull request by Maxfire**
### Description
Contact form module added, form created, config exported, tests added.

### CODE REVIEW

File: tests/src/ExistingSite/ContactFormTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    // User page creator.
+    $this->drupalLogin($this->createUser([
+      'create oe_showcase_page content',
+      'access corporate contact form',
+    ]));
+    $this->drupalGet('node/add/oe_showcase_page');
+
+    // Create oe_showcase_page.
+    $page->fillField('title[0][value]', 'Example contact form page');
```
Comment (brummbar) [inline tests/src/ExistingSite/ContactFormTest.php]: Quick note: please use labels and not paths, if possible. Use paths only for difficult nested elements that are repeated through the page (like when there are 3 fields "title").

File: modules/oe_showcase_contact_forms/config/install/field.storage.paragraph.field_contact_block.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - block_field
+    - paragraphs
+id: paragraph.field_contact_block
+field_name: field_contact_block
```
Comment (GilNovacomm) [inline modules/oe_showcase_contact_forms/config/install/field.storage.paragraph.field_contact_block.yml]: We've been using machine names for fields without field_ maybe we should also do this here to be consistent with other modules.

File: tests/src/ExistingSite/ContactFormTest.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+class ContactFormTest extends ShowcaseExistingSiteTestBase {
+
+  /**
+   * Check creation contact form content through the UI.
+   */
+  public function testCreateContactForm() {
+    $assert_session = $this->assertSession();
+    $page = $this->getSession()->getPage();
+
```
Comment (GilNovacomm) [inline tests/src/ExistingSite/ContactFormTest.php]: add statement at the beginning of the test to mark the created entities to be deleted:
 // Mark test content for deletion after the test has finished.
    $this->markEntityTypeForCleanup('node');
    $this->markEntityTypeForCleanup('paragraph');

**Review by GilNovacomm — DISMISSED** [review]
Ready for QA.

**Review by GilNovacomm — COMMENTED** [review]
Ready for QA

File: modules/oe_showcase_contact_forms/config/install/field.storage.paragraph.field_contact_block.yml (hunk 2 above)
Comment (drishu) [reply modules/oe_showcase_contact_forms/config/install/field.storage.paragraph.field_contact_block.yml]: that rule applies to oe components but its not needed in showcase

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.info.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+name: OpenEuropa Showcase Contact form
+type: module
+description: Provides a demo for the contact form.
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/oe_showcase_contact_forms.info.yml]:
```suggestion
description: Provides a demo for the contact form feature.
```

File: modules/oe_showcase_contact_forms/config/install/contact.form.example_contact_form.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    is_corporate_form: true
+    topic_label: ''
+    email_subject: 'Email subject'
+    header: ''
+    privacy_policy: 'entity:node/8'
+    includes_fields_in_auto_reply: false
+    allow_canonical_url: false
+    expose_as_block: true
+    optional_fields:
+      oe_country_residence: '0'
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/config/install/contact.form.example_contact_form.yml]:
```suggestion
      oe_country_residence: oe_country_residence
```

File: modules/oe_showcase_contact_forms/config/install/contact.form.example_contact_form.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      oe_telephone: '0'
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/config/install/contact.form.example_contact_form.yml]:
```suggestion
      oe_telephone: oe_telephone
```

File: tests/src/ExistingSite/ContactFormTest.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
+<?php
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]: Pls move this file into oe_showcase_contact_forms

File: tests/src/ExistingSite/ContactFormTest.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_showcase\ExistingSite;
+
+/**
+ * Contact form tests.
+ */
+class ContactFormTest extends ShowcaseExistingSiteTestBase {
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]:
```suggestion
class IntegrationTest extends ShowcaseExistingSiteTestBase {
```

IntegrationTest becasue we are testing the integration of the modules into showcase

File: tests/src/ExistingSite/ContactFormTest.php [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $entity_storage->load('example_contact_form');
+    $page->selectFieldOption(
+      'field_body[0][subform][contact_block][0][plugin_id]',
+      'oe_contact_forms_corporate_block:' . $contact_form->uuid(),
+    );
+    $page->pressButton('Save');
+
+    // Assert contact form was created successfully.
+    $assert_session->elementTextContains('css', '.alert-content', 'Example contact form page has been created.');
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]: remove this, these texts can change in time, rather go to the page and assert the labels are present

File: tests/src/ExistingSite/ContactFormTest.php [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    // User submitting the contact form.
+    $this->drupalLogin($this->createUser([
+      'access corporate contact form',
+    ]));
+
+    // Submit the form.
+    $this->drupalGet('/pages/example-contact-form-page');
+    $page->fillField('Subject', 'Example subject');
+    $page->fillField('Message', 'Example Message text');
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]: in config I asked for the phone and countries to be enabled, so we will fill also those in here (so your user will also need the 'View published SKOS Concept entities' permission)

File: tests/src/ExistingSite/ContactFormTest.php [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+    // Assert contact form was created successfully.
+    $assert_session->elementTextContains('css', '.alert-content', 'Example contact form page has been created.');
+
+    // User submitting the contact form.
+    $this->drupalLogin($this->createUser([
+      'access corporate contact form',
+    ]));
+
+    // Submit the form.
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]: remove all useless comments, please add comments only where code is harder to understand or ambiguous

File: tests/src/ExistingSite/ContactFormTest.php [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    $this->drupalGet('/pages/example-contact-form-page');
+    $page->fillField('Subject', 'Example subject');
+    $page->fillField('Message', 'Example Message text');
+    $page->selectFieldOption('Topic', 'Alpaca');
+    $page->checkField('privacy_policy');
+    $page->pressButton('Send message');
+
+    // Assert values were sent.
+    $assert_session->pageTextContains('Alpaca');
```
Comment (drishu) [inline tests/src/ExistingSite/ContactFormTest.php]: assert also the thank you message, not just the submission values;
and also assert the phone and country values here

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/ExistingSite/ContactFormTest.php (hunk 12 above)
Comment (Maxfire) [reply tests/src/ExistingSite/ContactFormTest.php]: Thank you message is not being asserted as I explained on OEL-1133, I could check the alert-success instead, but that is a change I proposed and is waiting to be merged on OEL-1178, so it needs to keep it as is for now.

Comment (drishu) [reply tests/src/ExistingSite/ContactFormTest.php]: "as I explained on OEL-1133" can you paste in a link pls

Comment (Maxfire) [reply tests/src/ExistingSite/ContactFormTest.php]: https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-1133#comment-5260348

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/IntegrationTest.php [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    $page->fillField('Subject', 'Example subject');
+    $page->fillField('Message', 'Example Message text');
+    $page->selectFieldOption('Topic', 'Alpaca');
+    $page->checkField('privacy_policy');
+    $page->fillField('Phone', '345345345');
+    $page->selectFieldOption('Country of residence', 'http://publications.europa.eu/resource/authority/country/BEL');
+    $page->pressButton('Send message');
+
+    $assert_session->pageTextContains('Alpaca');
+    $assert_session->pageTextContains('345345345');
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/tests/src/ExistingSite/IntegrationTest.php:61]: assert belgium
```
$assert_session->pageTextContains('Belgium');
```

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by GilNovacomm — APPROVED** [review]

**Pull request merged**